### PR TITLE
SAK-41262 'Publish Now' button should require confirmation

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: portal/portal-impl/impl/src/bundle/sitenav.properties [hunk 1]
```diff
@@ -52,6 +52,8 @@ sit_edit = Edit
 sit_reset = Tool Home
 sit_unpublished = Unpublished Site
 sit_publish_now = Publish Now
+sit_publish_confirmation = The site and its contents will be visible by the members, are you sure?
```
Comment (fostersdesign) [inline portal/portal-impl/impl/src/bundle/sitenav.properties]: This is a comma splice. Please change the comma to a period and capitalize the "A" in "are". 

Also, with the heading of this dialog as "Unpublished Site" even though it is about the publishing action, I think we should change the question to "Are you sure you want to publish this site?".

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm [hunk 2]
```diff
@@ -8,7 +8,25 @@
...
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal">&times;</button>
+                    <h4 class="modal-title">${rloader.sit_unpublished}</h4>
+                </div>
+                <div class="modal-body">
+                    <p>${rloader.sit_publish_confirmation}</p>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-default" data-dismiss="modal" onclick="publishSite('${sitePages.siteId}');">${rloader.sit_publish_now}</button>
```
Comment (fostersdesign) [inline portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: Should this be the primary button?

Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: Sure

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm [hunk 3]
```diff
@@ -8,7 +8,25 @@
...
+            <div class="modal-content">
+                <div class="modal-header">
+                    <button type="button" class="close" data-dismiss="modal">&times;</button>
+                    <h4 class="modal-title">${rloader.sit_unpublished}</h4>
+                </div>
+                <div class="modal-body">
+                    <p>${rloader.sit_publish_confirmation}</p>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-primary" data-dismiss="modal" onclick="publishSite('${sitePages.siteId}');">${rloader.sit_publish_now}</button>
```
Comment (mpellicer) [inline portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: @fostersdesign what are the classes for these buttons? these classes should be correct but doesn't work for me.

Comment (fostersdesign) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: .btn-primary should be correct. Is it being overridden by a library?

Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: For some reason, btn-primary doesn't work for me, this PR should be correct.

Comment (fostersdesign) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: I mocked up the situation and noticed that `button:not(.fc-button)` is causing `button.btn-primary` to not have enough specificity to make it look like the primary button. 

`button:not(.fc-button)` is such a broad rule that was added for the FullCalendar work done in the Lessons, I believe. I have noticed this issue in other cases that I've been working on in the recent past and have been thinking about removing it in favour of addressing the buttons in FullCalendar separately. 

So, I'm okay getting your change here into Sakai, @mpellicer. Then we can fix the `button:not(.fc-button)` separately.

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm [hunk 4]
```diff
@@ -8,7 +8,25 @@
 
     <div id="siteStatus" class="Mrphs-siteStatus Mrphs-messagePanel__information is-unpublished">
         <h4>${rloader.sit_unpublished}</h4>
-        <button onclick="publishSite('${sitePages.siteId}');" class="Mrphs-button--active">${rloader.sit_publish_now}</button>
+        <button class="Mrphs-button--active" data-toggle="modal" data-target="#publishSiteConfirmModal">${rloader.sit_publish_now}</button>
+    </div>
+
+    <div id="publishSiteConfirmModal" class="modal fade" role="dialog" tabindex="-1" aria-hidden="true">
```
Comment (fostersdesign) [inline portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: Is the `.fade` class necessary?

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm (hunk 3 above)
Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: good

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm (hunk 4 above)
Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: I tested here without the .fade and works too:
https://www.w3schools.com/bootstrap/bootstrap_modal.asp

Should we remove it?

Comment (fostersdesign) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/snippets/siteStatus-snippet.vm]: Removing it would be my preference, please.

**Pull request merged**